clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -3428,6 +3428,12 @@ void CXXNameMangler::mangleType(const BuiltinType *T) {
     Out << 'u' << type_name.size() << type_name;                               \
     break;
 #include "clang/Basic/AMDGPUTypes.def"
+#define HLSL_INTANGIBLE_TYPE(Name, Id, SingletonId)                            \
+  case BuiltinType::Id:                                                        \
+    type_name = #Name;                                                         \
+    Out << type_name.size() << type_name;                                      \
```

Comment:
```suggestion
    Out << 'u' << type_name.size() << type_name;                               \
```

The `u` is significant here because it denotes a vendor-specific builtin type (see: https://itanium-cxx-abi.github.io/cxx-abi/abi-mangling.html).